Apps/phrequesttracker/requesttracker_connector.py

```diff
@@ -0,0 +1,769 @@
+            self.save_progress("No attachments found for ticket id '{0}'".format(ticket_id))
+            return action_result.set_status(phantom.APP_SUCCESS)
+
+        # The first line of the attachments block starts with "Attachments: "
+        attachments[0] = attachments[0][13:]
+
+        # Get the attachment metadata
+        # Each line has the form "<attachment_id>: <file_name> (<content_type> / <size>)"
+        # The file name could have spaces and parentheses which makes this difficult
+        # But the filename will always be bordered by the first colon and the last open paren
+        for attachment in attachments:
+
+            data = {}
+
+            data['ticket_id'] = ticket_id
+            data['attachment_id'] = re.findall(r'\d+:', attachment)[0][:-1]
+
+            start_path = attachment.find(':') + 2
+            end_path = len(attachment) - attachment[::-1].find('(') - 2
+            data['path'] = attachment[start_path: end_path]
+
+            split_meta = attachment[end_path + 2:].split('/')
+            data['content_type'] = '{0}/{1}'.format(split_meta[0], split_meta[1].strip())
+            data['size'] = split_meta[2][:-1 if attachment.endswith(')') else -2].strip()
+
+            action_result.add_data(data)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _get_attachment(self, param):
+
+        # Create action result
+        action_result = self.add_action_result(ActionResult(param))
+
+        # Create RT object
+        if phantom.is_fail(self._create_rt_session(action_result)):
+            return action_result.get_status()
+
+        # Get parameters
+        ticket_id = param[RT_JSON_ID]
+        attachment_id = param[RT_JSON_ATTACHMENT]
+
+        # Set up the result data
+        data = action_result.add_data(dict())
+        data['id'] = ticket_id
+        data['attachment_id'] = attachment_id
+
+        # Request the attachment meta data
+        ret_val, resp_text = self._make_rest_call("ticket/{0}/attachments/{1}".format(ticket_id, attachment_id), action_result)
+
+        if phantom.is_fail(ret_val):
+            return ret_val
+
+        if '# Invalid attachment id' in resp_text:
+            return action_result.set_status(phantom.APP_ERROR, "Attachment with ID {0} not found on ticket {1}".format(attachment_id, ticket_id))
+
+        if re.findall('# Ticket .+ does not exist.', resp_text):
+            return action_result.set_status(phantom.APP_ERROR, "Ticket {0} does not exist.".format(ticket_id))
+
+        # Look for the filename in the meta data
+        filename_index = resp_text.find('filename')
+
+        if filename_index == -1:
+            file_name = "noname"
+            self.debug_print("Filename not found in header, setting to noname")
+        else:
+            start_filename = resp_text.find('=', filename_index) + 2
+            end_filename = resp_text.find('\n', filename_index) - 1
+            file_name = resp_text[start_filename: end_filename]
```

Comment:
can we do this with a regex? I don't know the exact format of the response (so it could require tweaking) but based on the current logic maybe something like:

```
filename_match = re.search(r'filename = ([^\n]+)', resp_text)
if not filename_match:
    ...
else:
    file_name = filename_match.group(1)
```